src/metrics/metrics.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#ifdef HAVE_PROMETHEUS
+#include <prometheus/counter.h>
+#include <prometheus/exposer.h>
+#include <prometheus/registry.h>
+#endif
+
+namespace metrics {
+
+class MetricCounter {
+public:
+#ifdef HAVE_PROMETHEUS
+	explicit MetricCounter(prometheus::Counter *counter = nullptr) : counter_(counter) {}
+
+	inline void increment(double n = 1) const {
+		if (counter_ != nullptr) {
+			counter_->Increment(n);
+		}
+	}
+
+private:
+	prometheus::Counter *counter_{};
+#else
+	// Dummy class for packages without prometheus
+	explicit MetricCounter() = default;
+
+	inline void increment(double = 1) const {
+	}
+#endif
+};
+
+void metrics_init(const char* host);
+void metrics_destroy();
+
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+
+extern MetricCounter rx_client_packets;
+extern MetricCounter tx_client_packets;
```

Comment:
Why extern here?

Reply:
I need other parts of master to access these variables, but if I define them here then it's (potentially) a ODR violation. Hence external here and definition in .cc

Reply:
Are inline variables not enough here? My concern is about later usage in shared libraries and plugins, we can end up with variables pointing to different places.

Reply:
Inline necessitates a lock:

`extern`:
https://godbolt.org/z/Ync9cPTEv

`inline`:
https://godbolt.org/z/zx963xx33

I'll think about redesigning this a bit, maybe use a switch statement instead of variables

Reply:
I redesigned how these variables are accessed. It now uses array subscripting with enums instead.